update/genesis/import.go

```diff
@@ -80,47 +79,51 @@ func NewStateImport(args ArgsNewStateImport) (*stateImport, error) {
 		trieStorageManagers: args.TrieStorageManagers,
 		storageConfig:       args.StorageConfig,
 		shardID:             args.ShardID,
+		hardforkStorer:      args.HardforkStorer,
 	}
 
 	return st, nil
 }
 
 // ImportAll imports all the relevant files for the new genesis
 func (si *stateImport) ImportAll() error {
-	files := si.reader.GetFileNames()
-	if len(files) == 0 {
-		return update.ErrNoFileToImport
-	}
-
-	var err error
-	for _, fileName := range files {
-		switch fileName {
-		case MetaBlockFileName:
-			err = si.importMetaBlock()
-		case MiniBlocksFileName:
-			err = si.importMiniBlocks()
-		case TransactionsFileName:
-			err = si.importTransactions()
+	var errFound error
+
+	si.hardforkStorer.RangeKeys(func(identifier string, keys [][]byte) {
+		var err error
```

Comment:
define this function separately and you have one liner here.

Reply:
can not do that because of errFound